### ISSUE TITLE
Flake: `TestGetStateRPC`

### ISSUE THREAD

**Issue by balopat** [GoogleContainerTools/skaffold#2100]
```
time="2019-05-09T22:32:42Z" level=info msg="Ran in 6.011332858s"
--- FAIL: TestGetStateRPC (7.12s)
    rpc_test.go:286: waiting for connection...
    rpc_test.go:246: skaffold build or deploy not complete. state: buildState:<artifacts:<key:"gcr.io/k8s-skaffold/test-dev" value:"In Progress" > > deployState:<status:"Not Started" >
```

https://source.cloud.google.com/results/invocations/da264594-6ab4-48fe-b761-e8510853853a/targets/runtimes-common%2Fskaffold%2Fpresubmit/log

**Comment by dgageot** [comment]
Let's close this one since it hasn't happened for a while

**Comment by tejal29** [comment]
Saw this again on my a my PR. 

```
--- FAIL: TestGetStateRPC (0.82s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x140e3fc]
goroutine 741 [running]:
testing.tRunner.func1(0xc0000e8300)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x1590da0, 0x277bc00)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/GoogleContainerTools/skaffold/integration.checkBuildAndDeployComplete(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000686a80)
	/skaffold/integration/rpc_test.go:370 +0x2c
github.com/GoogleContainerTools/skaffold/integration.TestGetStateRPC(0xc0000e8300)
	/skaffold/integration/rpc_test.go:268 +0x465
testing.tRunner(0xc0000e8300, 0x180d550)
	/usr/local/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a
FAIL	github.com/GoogleContainerTools/skaffold/integration	462.053s
```

**Comment by dgageot** [comment]
Fixed with #3605
